### PR TITLE
History Search

### PR DESCRIPTION
**Pull request by pureooze**
- Enables history search in Saka
- Prioritizes most frequently visited sites over most recently visited

### CODE REVIEW

**Comment by eejdoowad** [comment]
Great work. Long term, it might be worthwhile exploring the approach [FauxBar](https://github.com/ChrisNZL/Fauxbar) takes. It provides more powerful search by maintaining its own index of browser history in a websql database.

**Comment by pureooze** [comment]
Let's be honest you did most of the work, I just uncommented it 😃

File: src/suggestion_engine/server/providers/closedTab.js [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
   const sessions = await browser.sessions.getRecentlyClosed();
   const filteredSessions = [];
 
+  // TODO: This for loop is currently flagged by the airbnb eslint rules.
+  // See: https://github.com/airbnb/javascript/issues/1271
+  // Not disabling the rule as this might be fixable in the future using filter.
+  // This for loop is needed at the moment as a workaround since filter does not support async.
   for (const session of sessions) {
     const sakaUrl = await isSakaUrl(session.tab.url);
     if (session.tab && !sakaUrl) {
```
Comment (eejdoowad) [inline src/suggestion_engine/server/providers/closedTab.js]: browser.sessions.getRecentlyClosed() returns a list of closed tabs AND WINDOWs. This if statement filters the windows out. Ideally, it should be possible to restore windows, but I removed them to simplify the implementation.

Comment (pureooze) [reply src/suggestion_engine/server/providers/closedTab.js]: Good point, I will make a note to look into this. I skimmed through the documentation for [sessions](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/sessions/restore) and this doesn't seem too difficult to implement, there just needs to be a check to see if it is a tab or window and pass in the appropriate value.

**Pull request merged**